ownedbytes/src/lib.rs

```diff
@@ -139,6 +139,16 @@ impl OwnedBytes {
         self.advance(8);
         u64::from_le_bytes(octlet)
     }
+
+    /// Reads an `u32` encoded as little-endian from the `OwnedBytes` and advance by 4 bytes.
+    #[inline]
+    pub fn read_u32(&mut self) -> u32 {
+        assert!(self.len() > 3);
```

Comment:
Doesn't `as_slice()[..4]` basically assert the same condition?

Reply:
It does, I copied from `read_u64`, not sure why it does that check either